plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt

```diff
@@ -107,3 +113,94 @@ class SourcesetDependencyAppender(val context: DokkaContext) : PageTransformer {
         )
     }
 }
+
+class SearchbarDataInstaller(val context: DokkaContext): PageTransformer {
+    private val pageList = mutableMapOf<String, Pair<String, String>>()
+    lateinit var locationProvider : LocationProvider
+
+        override fun invoke(input: RootPageNode): RootPageNode {
+            locationProvider = context.plugin<DokkaBase>().querySingle { locationProviderFactory }
```

Comment:
I am not so sure about fetching location provider like that. It requires a root page that can be changed in other `PageTransformer`, so unless you are entirely sure that this page transformer is last i'd just create this object in HtmlRenderer and pass location provider via constructor

Reply:
I've changed usage of SearchbarDataInstaller - I don't declare it as PageTransformer and use it in HtmlRenderer to register and finally to generate page list.